test/bucket_test.ts

```diff
@@ -42,26 +44,26 @@ describe("Bucket", () => {
 
   /** @test {Bucket#getData} */
   describe("#getData()", () => {
-    it("should execute expected request", () => {
+    it("should execute expected request", async () => {
       const executeStub = sandbox
         .stub(client, "execute")
-        .returns(Promise.resolve());
+        .returns(Promise.resolve({ headers: fakeHeaders }));
 
-      getBlogBucket().getData();
+      await getBlogBucket().getData();
 
       sinon.assert.calledWithMatch(executeStub, {
         path: "/buckets/blog",
         headers: {},
       });
     });
 
-    it("should resolve with response data", () => {
+    it("should resolve with response data", async () => {
       const response = { data: { foo: "bar" }, permissions: {} };
       sandbox.stub(client, "execute").returns(Promise.resolve(response));
 
-      return getBlogBucket()
-        .getData()
-        .should.become({ foo: "bar" });
+      ((await getBlogBucket().getData()) as { foo: string }).should.deep.equal({
```

Comment:
Maybe in this case, an intermediary variable would make sense

Reply:
I've updated the tests to use an intermediary variable whenever we were casting an awaited value. However, we still have tests that look like this:

```ts
(await api.fetchServerInfo()).should.deep.equal(fakeServerInfo);
```

Is this clear enough, or should we lift all awaited calls into variables?

Reply:
Yes, this was like one of the few complex ones. Thanks for the changes!